### PR TITLE
Target Winver XP causes problems in Windows 10

### LINKED ISSUE #294: Strange characters in console - only from this installer builds
### Steps to reproduce

1. Run the installer, full options
2. open up cmd.exe,
3. start; vim --clean -U NONE
4. open help;  h:
5. use mouse visual select some text, and drag the visual selection around a bit.
6. see weird characters;
image

### Expected behaviour

Expect no weird characters.

I couldn't get it to occur when I compile the same patch levels myself.  I only see the problem with these appveyor builds.

I copied the appveyor.bat file, and tried to reproduce the build as close as possible to what appveyor is doing.  But, I did have to make a couple of changes;

Python version to 3.11.1
set PYTHON3_VER=311
set PYTHON3_RELEASE=3.11.1

Ruby version from 3.0 to 3.2
set RUBY_VER=32
set RUBY_API_VER_LONG=3.2.0
set RUBY_BRANCH=ruby_3_2
set RUBY_RELEASE=3.2.0-1

and some manual steps to get ruby to compile and install - the configure step was failing due to unfound header files.

The only other difference I could think of, is that appveyor.yml file mentions Visual Studio 2015, but I'm using Visual Studio 2022.

I'll install increasingly older versions until I find which one it starts at.  Will update what I find.

### Version of Vim

gvim_9.0.1097_x64.exe, gvim_9.0.1117_x64.exe,  and gvim_9.0.1122_x64.exe

### Environment

Windows 11 21H2

### PR DESCRIPTION
**Pull request by zewpo**
This PR drops the XP target from the build, because that causes weird characters in the console on newer versions of windows.

Potentially Closes #294

### CODE REVIEW

File: appveyor.bat [hunk 1]
```diff
@@ -289,14 +276,14 @@ cd vim\src
...
-start /wait cmd /c ""C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd" /x64 && cd GvimExt && nmake -f Make_mvc.mak CPU=AMD64 clean all > ..\gvimext.log"
+start /wait cmd /c "cd GvimExt && nmake -f Make_mvc.mak CPU=AMD64 clean all > ..\gvimext.log"
 type gvimext.log
 copy GvimExt\gvimext.dll   GvimExt\gvimext64.dll
 move GvimExt\gvimext.dll   GvimExt64\gvimext.dll
 copy /Y GvimExt\README.txt GvimExt64\
 copy /Y GvimExt\*.inf      GvimExt64\
 copy /Y GvimExt\*.reg      GvimExt64\
-start /wait cmd /c ""C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd" /x86 && cd GvimExt && nmake -f Make_mvc.mak CPU=i386 clean all > ..\gvimext.log"
+start /wait cmd /c "cd GvimExt && nmake -f Make_mvc.mak CPU=i386 clean all > ..\gvimext.log"
```
Comment (k-takata) [inline appveyor.bat:286]: Only setting `CPU` doesn't work.
We need to build 32-bit gvimext even if we build 64-bit vim, and we need to build 64-bit gvim even if we build 32-bit vim. So, we should call `vcvarsall.bat` for each architecture.

Comment (zewpo) [reply appveyor.bat:286]: ok, thanks. @k-takata 

note - I'm getting the test failures again using VS 2022. runner image 
I previously thought that targeting windows XP was causing this, 
just bactracing my steps now to work it out.

**Comment by zewpo** [comment]
Does appveyor timeout and abort if stuck, or just hang?

Build seems stuck for 52+ minutes, and counting...

https://ci.appveyor.com/project/chrisbra/vim-win32-installer/builds/45856402/job/i8v4r8aniwbvs2t4

stuck on;
..\gvim -u NONE -U NONE --noplugin --not-a-term --gui-dialog-file guidialog -S runtest.vim test_autocmd.vim

It's holding up the queue.

**Comment by zewpo** [comment]
Good news is that the installer is still working fine on Windows 7.  Also, this build of vim is passing all the test scripts.

Bad news is that this build still displays the weird characters in windows 11 console.  The whole point of this PR was meant to fix that, so.... not there yet...

And, I get a different problem when building with VS 2022 builder image,  the weird characters are no longer displayed,  So, it seems to fix that....     BUT, the GUI starts to fail on some of the test cases.    Now, strange thing is, that building VIM in github CI, VS 2022, it passes all of those same GUI tests.   (Or... does it skip those tests - I will triple check that)

One thing I noted, is that the github CI seems to use Windows Server 2022 OS for its VS 2022 builder image.    While, appveyor uses Windows Server 2019 OS for its VS 2022 builder image.

**Comment by k-takata** [comment]
It looks that you tried to disable/enable Racket. Did it make difference?
Racket is not tested in the main repository (on Windows).

**Comment by zewpo** [comment]
@k-takata   yeah, nah, ruled it out, no difference - the GUI  test [Test_gui_lowlevel_keyevent]  keeps failing with or without racket.

now, strange thing happens, when I make the build progress to packaging by comment out testing, to skip the problem, then I can download the artifacts from appveyor, and run the same test scripts against them that would have failed if they weren't commented out in appveyor,  but they pass on my own computer.

**Comment by k-takata** [comment]
Hmm. Does it depend whether we use winpty or ConPTY?

**Comment by zewpo** [comment]
@k-takata   possibly.     Although, it's a GUI test that's failing.  The console is fine, but I can check anyway - something is different.   I'm also trying to boot up an older windows 10 vm to test it there.

**Comment by zewpo** [comment]
b.t.w. @k-takata   Confirming that I checked with winpty vs conpty.  No difference - still can't reproduce the issue anywhere except for inside appveyor's test environment.  

Umm, hang on... I've not tried with just conpty in appveyor test environment.  Let me give that a go and see what happens...

**Comment by zewpo** [comment]
Also, checked that dropping WinPTY does not make the Windows XP problem go away.  ie. I still get the garbled characters from the build that targets XP - even when I dont include winpty in the build.

**Comment by zewpo** [comment]
Coming back to this;

> Bad news is that this build still displays the weird characters in windows 11 console. The whole point of this PR was meant to fix that, so.... not there yet...

So, dropping XP did NOT actually resolve this issue of garbled characters.   It happens less often, not so repeatable, but it still happens intermittently.  

So, I think I will close this PR.    There may be other solutions possible - but I've checked a number of times, and upgrading to VS 2022 was a reliable way I found to fix this issue. See PR #296

**Pull request closed without merging**